tutorials/block-texture-variation.md

```diff
@@ -15,17 +15,64 @@ parent: Tutorials
 {:toc}
 </details>
 
-Beginner
-{: .label .label-green }
+intermediate  
+{: .label .label-yellow }
 
 Block texture variation is when a single block can have multiple textures.  This is useful for blocks such as dirt or grass where some blocks may have slight variations such as small rocks and others don't.
 
-To enable texture variations you create a folder with the block name in the textures / blocks resource pack folder.  You then store your texture variations as numbered png images, starting at 0.
+To enable texture variations you create a terrain_texture.json file in the textures folder. The terrain_texture.json in the block definition the texture should be set to a dictionary with a variation key, the varation key is an array of dictionaries that must contain a path key pointing to the texture file and may contain a weight to control how often the textures show up.
 
 ## Example Dirt Texture Variation
 
 This is an example on how to create texture variations for the dirt block with three images:
-- Create a `textures/blocks/dirt` folder in the resource pack
-- Create or modify three dirt textures, name them `0.png`, `1.png`, and `2.png`
-- Copy the `0.png`, `1.png`, and `2.png` to the `textures/blocks/dirt` folder
-- Remove `textures/blocks/dirt.png` if it exists
\ No newline at end of file
+- Create a `textures/terrain_texture.json` file in the resource pack
+- in the json file you need to define the blocks that have variation as shown below
+'''json
```

Comment:
These look a like like ' not `. Also should be ```jsonc, not '''json.

If you don't respond in a day or two I will just merge it in, and fix it afterwards, but it would be nice to update the PR if you could.

Reply:
I believe it is fixed now.

Reply:
Many thanks for the nice PR!